ptychodus/model/ptychonn/reconstructor.py

```diff
@@ -105,12 +104,24 @@ def __init__(self, modelSettings: PtychoNNModelSettings,
     def name(self) -> str:
         return 'AmplitudePhase' if self._enableAmplitude else 'PhaseOnly'
 
-    def _createModel(self) -> ReconSmallModel:
+    def _createModel(self) -> ptychonn.LitReconSmallModel:
         logger.debug('Building model...')
-        return ReconSmallModel(
-            nconv=self._modelSettings.numberOfConvolutionKernels.value,
-            use_batch_norm=self._modelSettings.useBatchNormalization.value,
-            enable_amplitude=self._enableAmplitude,
+        if self._modelSettings.stateFilePath.value.exists():
+            path = self._modelSettings.stateFilePath.value
+            parameters = None
+        else:
+            path = None
+            parameters = dict(
+                nconv=self._modelSettings.numberOfConvolutionKernels.value,
+                use_batch_norm=self._modelSettings.useBatchNormalization.value,
+                enable_amplitude=self._enableAmplitude,
+                max_lr=float(self._trainingSettings.maximumLearningRate.value),
+                min_lr=float(self._trainingSettings.minimumLearningRate.value),
+            )
```

Comment:
The behavior of createModel is now that a model is created if the stateFilePath does not exist, otherwise it tries to load the model from the existing file. Not sure if that was the previous behavior. This means that a model is always returned, but if the user provided an incorrect model path, they will get a randomly initialized model without any error.